### PR TITLE
Task 5

### PR DESCRIPTION
**Pull request by tarunbhartiya7**
### Tasks Completed:
- File serverless.yml contains configuration for importProductsFile function
- The importProductsFile lambda function returns a correct response which can be used to upload a file into the S3 bucket
- Frontend application is integrated with importProductsFile lambda
- The importFileParser lambda function is implemented and serverless.yml contains configuration for the lambda

### Additional Tasks Completed:
- async/await is used in lambda functions
- At the end of the stream the lambda function should move the file from the uploaded folder into the parsed folder (move the file means that file should be copied into a new folder in the same bucket called parsed, and then deleted from uploaded folder)

`Cloudfront`: https://d1rxysx81o0hyh.cloudfront.net

### Screenshots:

`generate presigned url and upload in s3`

image

`read products.csv and log contents in cloudwatch`

image

### CODE REVIEW

**Comment by ilias-bikenov** [comment]
Reviewed. Score is 6/8

File: import-service/importFileParser.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const AWS = require("aws-sdk")
+const s3 = new AWS.S3()
+require("../config")
+const csv = require("csv-parser")
+
+module.exports.importFileParser = async (event) => {
+  const params = {
+    Bucket: process.env.S3_BUCKET,
+    Key: `uploaded/products.csv`,
```
Comment (SlavaJSFE) [inline import-service/importFileParser.js:9]: You should get a file name from the event

File: serverless.yml [hunk 2]
```diff
@@ -13,33 +13,55 @@ provider:
...
           cors: true
   createProduct:
-    handler: products.createProduct
+    handler: product-service.createProduct
     events:
       - http:
           path: /products
           method: post
           cors: true
+  importProductsFile:
```
Comment (SlavaJSFE) [inline serverless.yml:45]: According to the task you should have separated serverless.yml files for each service.

**Review by SlavaJSFE — APPROVED** [review]

**Comment by SlavaJSFE** [comment]
Score 7

File: import-service/importFileParser.js (hunk 1 above)
Comment (tarunbhartiya7) [reply import-service/importFileParser.js:9]: Noted. I will make these changes in another PR.

File: serverless.yml (hunk 2 above)
Comment (tarunbhartiya7) [reply serverless.yml:45]: Noted. I will make these changes in another PR.

**Pull request merged**